data/the emerald sword.txt

```diff
@@ -0,0 +1,796 @@
+			choice
+				`(Say nothing.)`
+					goto middle
+				`"Are you a..."`
+			`	They cut you off. "Who or what I am doesn't matter to you."`
+			label middle
+			`	"Let's get to what matters. You want to find that alien artifact, yes?"`
+			`	You nod in response.`
+			`	"I have the information you need. Finding lost objects is the most trivial of my services, but as with everything, it has a price.`
+			`	"Ten million credits." they say. "Non-negotiable. And don't try anything funny."`
+			branch cash
+				credits >= 12000000
+			choice
+				`	"What? I don't have that much money!"`
+				goto nomoney
+				`	"That's expensive. If it's such a trivial service, can't I have a discount?"`
+					goto haggle
+				`	"How about you tell me what I want and I don't blow up your ship and your pirate friends?"`
+					goto threat
+			label visited
+			`You come back to Stormhold and receive new coordinates to meet with Mammon. Would you like to visit them?`
+			choice
+				`	(Yes.)`
+				`	(No.)`
+					defer
+			`	You follow the new coordinates, arriving again in a random place in the sea, again under stormy conditions, which seems to be the norm in this planet. The submarine emerges, you land, and follow the men to Mammon's room.`
+			`	"Welcome back," they say.`
+			label question
+			`	"I hope you are ready to make your payment."`
+			branch cash
+				credits >= 12000000
+			choice
+				`	"I don't have that much money to spare."`
+					goto nomoney
+				`	"Are you ready to give me a discount?"`
+					goto haggle
+				`	"How about you tell me what I want and I don't blow up your ship and your pirate friends?"`
+					goto threat
+			label cash
+			choice
+				`	"I need to think about this. Give me some time."`
+					defer
+				`	"That's a lot of money just for information. Can you make it cheaper?"`
+					goto haggle
+				`	"How about you tell me what I want and I don't blow up your ship and your pirate friends?"`
+					goto threat
+				`	"Here's the money."`
+					goto pay
+			label nomoney
+			`	"I know you don't. I can arrange for the Payment to be made in different ways. Organ donation, prostitution, volunteering as a test subject... We'll need an appraisal first though."`
+			`	You politely decline.`
+			`	"Come back to me when you have it, then, and I shall provide you with the information. We have all the time in the universe. Until you die, or someone decides to destroy the cube, that is."`
+			`	They gesture, and the men promptly and diligently escort you back out to your ship.`
+				defer
+			label haggle
+			`	Mammon doesn't look pleased.`
+			`	"What part of non-negotiable did you not understand? You're pushing your luck, captain <last>."`
+				goto question
```

Comment:
This allows an infinite loop to be formed where the player can continue haggling with no effect. I'd recommend changing it so that you can haggle once and then have to make an actual decision.

Reply:
Good idea, i'll try this.